Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Find.swift

```diff
@@ -219,10 +212,21 @@ extension PathHierarchy {
         var node = startingPoint
         var remaining = pathComponents[...]
         
+        /// Ensure that the return value is valid for this search. 
+        func validateReturnValue(_ node: Node) throws -> Node {
+            if node.identifier == nil {
+                throw Error.unfindableMatch(node)
+            }
```

Comment:
Why is this check needed if the `identifier` property inside the`Node` class is an implicitly unwrapped optional?
```swift
fileprivate(set) var identifier: ResolvedIdentifier!
```
If there's a possibility for this value to be nil, then shouldn't it be an optional value?

All the tests pass without this check, so I'm not clear on whether it's needed.